### DISCUSSION TITLE
Default Augmentation when using the Train Mode

### DISCUSSION THREAD

**Discussion by NiklasHansen1993** [ultralytics/ultralytics#23611] [General]
Hello everyone, 
i'm very new to using Objectdetection and anything to do with coding in general. I've been training a few models on datasets consisting of Blood smear images in order to detect white blood cells. When starting the training with IMGSZ = 640
BATCH = 16
EPOCHS = 300
PATIENCE = 50
i get a long list of values for different augmentation arguments - does YOLO automatically apply augmentation for training if nothing else is specified? And is this reccommended? 

Thank you in advance, Niklas

**Comment by UltralyticsAssistant** [comment]
👋 Hello @NiklasHansen1993, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get started, and an Ultralytics engineer will also assist you soon 😊. We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question (like your augmentation/training-settings question), please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). In particular, please paste the portion of your training output where the augmentation arguments are printed, plus your exact train command/script and `ultralytics` version.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by glenn-jocher** [reply]
Yes—if you don’t specify augmentation args, Ultralytics YOLO will train with the default augmentation settings (and prints them at startup); that’s generally recommended as a baseline, but for microscopy/blood-smear data you may want to dial back “scene-changing” transforms (especially `mosaic` and strong geometry) if they create unrealistic samples—see the [YOLO augmentation guide](https://docs.ultralytics.com/guides/yolo-data-augmentation/) and the [training augmentation arguments table](https://docs.ultralytics.com/modes/train/).

```python
from ultralytics import YOLO

model = YOLO("yolo26n.pt")
model.train(
    data="your_dataset.yaml",
    imgsz=640,
    batch=16,
    epochs=300,
    patience=50,
    mosaic=0.0,      # often worth trying for microscopy
    mixup=0.0,
    degrees=0.0,
    translate=0.0,
    shear=0.0,
    perspective=0.0,
    # keep mild color aug if staining/illumination varies:
    hsv_h=0.01,
    hsv_s=0.2,
    hsv_v=0.2,
)
```

**Reply by NiklasHansen1993** [reply]
perfect, thank you so much, i will try that! I'm also trying to test the models performance when training with a smaller and smaller amount of training data - do you have any tips for that regarding the hyperparameters? Or are there legit ways of using Augmentations to enlarge a small dataset that only consists of 10 - 50 pictures by copying part of the training data and using augmentations on these copies?

**Reply by glenn-jocher** [reply]
You don’t need to physically copy images to “grow” the dataset—Ultralytics YOLO applies augmentations on-the-fly each epoch, so a 10–50 image set effectively produces many unique variants during training; for low-data experiments, keep one fixed held-out test/val split (don’t shrink it each run) and expect high variance, so train a few repeats and prefer a smaller model like `yolo26n.pt` to reduce overfitting, then only add augmentations that still look like real blood-smear images (more isn’t always better). If you want stronger/medical-specific variation beyond the built-ins, you can pass custom Albumentations transforms via the Python API as shown in the [Albumentations integration guide](https://docs.ultralytics.com/integrations/albumentations/) (this augments in-memory during training; it’s not creating saved copies), and the general idea is summarized in the [data augmentation glossary](https://www.ultralytics.com/glossary/data-augmentation).
